File: Leetcode/C++/65. Valid Number/solution.cpp
// https://leetcode.com/problems/valid-number/

class Solution
{
public:
    bool isNumber(string s)
    {
        static const regex re("^([+-]?((\\d+\\.\\d*)|(\\.\\d+)|(\\d+))([eE][+-]?\\d+)?)$");

        return regex_match(s, re);
    }
};

File: Leetcode/Python/65. Valid Number/solution.py
# https://leetcode.com/problems/valid-number/

import re


class Solution:
    def isNumber(self, s: str) -> bool:
        p = re.compile("^([+-]?((\d+\.\d*)|(\.\d+)|(\d+))([eE][+-]?\d+)?)$")

        if p.match(s):
            return True
        else:
            return False
